### PR TITLE
rnaseq pipeline config updated from biofx team input

### PR DESCRIPTION
**Pull request by curlup**

### CODE REVIEW

File: cidc_schemas/prism/pipelines.py [hunk 1]
```diff
@@ -131,13 +133,32 @@ def _rnaseq_pipeline_config(full_ct: dict, patch: dict, bucket: str) -> Dict[str
 
     # as we know that `patch` is a prism result of a rna_fastq upload
     # we are sure these getitem calls should be fine
-    new_data = [r for assay in patch["assays"]["rna"] for r in assay["records"]]
-
+    # and that there should be just one rna assay
+    assay = patch["assays"]["rna"][0]
+
+    # splitting samples from different participants into different groups
+    groups = defaultdict(list)
```
Comment (jacoblurye) [inline cidc_schemas/prism/pipelines.py:140]: `participant_groups`, maybe, to hint at what the keys represent?

File: cidc_schemas/prism/pipelines.py [hunk 2]
```diff
@@ -131,13 +133,32 @@ def _rnaseq_pipeline_config(full_ct: dict, patch: dict, bucket: str) -> Dict[str
...
+        all_pids.append(pid)
+        all_cimac_ids.append(record["cimac_id"])
+        groups[pid].append(record)
+
+    dt = datetime.now().isoformat(timespec="minutes").replace(":", "-")
     res = {}
-    for sample_data in new_data:
-        res[sample_data["cimac_id"] + ".yaml"] = templ.render(
-            BIOFX_BUCKET_NAME=bucket, **sample_data
+    # TODO add f"metasheet_{dt}.csv" generation with all samples/participants metadata
```
Comment (jacoblurye) [inline cidc_schemas/prism/pipelines.py:151]: This isn't done below?

File: cidc_schemas/prism/pipelines.py [hunk 3]
```diff
@@ -131,13 +133,32 @@ def _rnaseq_pipeline_config(full_ct: dict, patch: dict, bucket: str) -> Dict[str
...
+        config_str = templ.render(
+            BIOFX_BUCKET_NAME=bucket,
+            samples=samples,
+            paired_end_reads=assay["paired_end_reads"],
         )
+        # keying on participant id and date time, so if data for one participant
+        # comes in different uploads, those runs will be distinguishable
+        res[f"rna_pipeline_{pid}_{dt}.yaml"] = config_str
 
     return res
```
Comment (jacoblurye) [inline cidc_schemas/prism/pipelines.py:163]: The docstring for this function doesn't accurately describe its actual return type.

File: cidc_schemas/prism/pipelines.py (hunk 2 above)
Comment (curlup) [reply cidc_schemas/prism/pipelines.py:151]: nah, they want a separate .csv sheet with metadata.

File: cidc_schemas/prism/pipelines.py (hunk 3 above)
Comment (curlup) [reply cidc_schemas/prism/pipelines.py:163]: Good catch, thanks)

File: cidc_schemas/schemas/assays/components/ngs/rna/star.json [hunk 4]
```diff
@@ -0,0 +1,80 @@
+{
+  "$schema": "metaschema/strict_meta_schema.json#",
+  "$id": "star",
+  "title": "RNAseq Star Output Files",
+  "type": "object",
+  "description": "Star output sample files.",
+  "properties": {
+    "chimeric_out_junction": {
+      "$ref": "artifacts/artifact_text.json"
```
Comment (roshni-b) [inline cidc_schemas/schemas/assays/components/ngs/rna/star.json:9]: New artifact artifact_junction.json?

File: cidc_schemas/schemas/assays/components/ngs/rna/salmon.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+          "$ref": "artifacts/artifact_text.json"
+      },
+      "aux_info_observed_bias_gz": {
+          "$ref": "artifacts/artifact_gz.json"
+      },
+      "aux_info_observed_bias_3p_gz": {
+          "$ref": "artifacts/artifact_gz.json"
+      },
+      "cmd_info_json": {
+          "$ref": "artifacts/artifact_text.json"
```
[the remark this answers is not in the document]
Comment (curlup) [reply cidc_schemas/schemas/assays/components/ngs/rna/salmon.json:33]: Thanks for catching this!

File: cidc_schemas/schemas/assays/components/ngs/rna/star.json (hunk 4 above)
Comment (curlup) [reply cidc_schemas/schemas/assays/components/ngs/rna/star.json:9]: I honestly don't want to introduce new kinds of artifacts for all that.
I'm starting to think that such granularity of artifact schemas is unasked for.

**Pull request merged**